subsys/net/lib/shell/http.c

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(net_shell);
+
+#include "net_shell_private.h"
+#include <zephyr/net/http/service.h>
+#include <zephyr/net/http/server.h>
+
+static int cmd_net_http(const struct shell *sh, size_t argc, char *argv[])
+{
+#if defined(CONFIG_HTTP_SERVER)
+	int res_count = 0, serv_count = 0;
+
+	PR("%-15s\t%-12s\n",
+	   "Host:Port", "Concurrent/Backlog");
+	PR("\tResources\n");
+	PR("\n");
+
+	HTTP_SERVICE_FOREACH(svc) {
+		PR("%s:%d\t%zu/%zu\n",
+		   svc->host == NULL || svc->host[0] == '\0' ?
+		   "<any>" : svc->host, svc->port ? *svc->port : 0,
+		   svc->concurrent, svc->backlog);
+
+		HTTP_SERVICE_FOREACH_RESOURCE(svc, res) {
+			PR("\t%s\n", res->resource);
```

Comment:
Maybe we could fetch the resource detail, and print the type of resource here as well?

Reply:
Good idea. Now the shell prints this
```
Host:Port       Concurrent/Backlog
        Resource type   Methods         Endpoint

192.0.2.1:80    1/10
             dynamic    GET,POST        /dynamic
              static    GET             /
           websocket    GET             /

1 service and 3 resources found.
```